src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -34,7 +34,8 @@ trait WorkflowManager {
    * Generates a workflow with an ID, inserts it into the store and returns the pair.
    */
   def submitWorkflow(wdl: WdlSource, inputs: WorkflowInputs): Try[ManagedWorkflow] = {
-    val managedWorkflow = generateWorkflow(wdl, inputs) map {x => ManagedWorkflow(UUID.randomUUID(), x)}
+    val workflowId = UUID.randomUUID()
+    val managedWorkflow = generateWorkflow(workflowId, wdl, inputs) map {x => ManagedWorkflow(workflowId, x)}
     // If we successfully generated a workflow and successfully inserted it into the store, return the workflow/id combo
     for (wf <- managedWorkflow; m <- workflowStore.insert(wf.id, wf.workflow)) yield wf
```

Comment:
@mcovarr My attempts at forcomping the whole thing always ran afoul of the underlying magic. I might be missing some obvious way of making this series of statements more simple but everything I tried ended up looking crappier to my eyes. I'm inclined to leave it unless you really hates it or have something specific.